src/components/views/rooms/ReplyTile.js

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2020 Tulir Asokan <tulir@maunium.net>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { _t, _td } from '../../../languageHandler';
+
+import * as sdk from '../../../index';
+
+import dis from '../../../dispatcher/dispatcher';
+import SettingsStore from "../../../settings/SettingsStore";
+import {MatrixClient} from 'matrix-js-sdk';
+
+import { objectHasDiff } from '../../../utils/objects';
+
+const eventTileTypes = {
+    'm.room.message': 'messages.MessageEvent',
+    'm.sticker': 'messages.MessageEvent',
+    'm.call.invite': 'messages.TextualEvent',
+    'm.call.answer': 'messages.TextualEvent',
+    'm.call.hangup': 'messages.TextualEvent',
+};
+
+const stateEventTileTypes = {
+    'm.room.aliases': 'messages.TextualEvent',
+    // 'm.room.aliases': 'messages.RoomAliasesEvent', // too complex
+    'm.room.canonical_alias': 'messages.TextualEvent',
+    'm.room.create': 'messages.RoomCreate',
+    'm.room.member': 'messages.TextualEvent',
+    'm.room.name': 'messages.TextualEvent',
+    'm.room.avatar': 'messages.RoomAvatarEvent',
+    'm.room.third_party_invite': 'messages.TextualEvent',
+    'm.room.history_visibility': 'messages.TextualEvent',
+    'm.room.encryption': 'messages.TextualEvent',
+    'm.room.topic': 'messages.TextualEvent',
+    'm.room.power_levels': 'messages.TextualEvent',
+    'm.room.pinned_events': 'messages.TextualEvent',
+    'm.room.server_acl': 'messages.TextualEvent',
+    'im.vector.modular.widgets': 'messages.TextualEvent',
+    'm.room.tombstone': 'messages.TextualEvent',
+    'm.room.join_rules': 'messages.TextualEvent',
+    'm.room.guest_access': 'messages.TextualEvent',
+    'm.room.related_groups': 'messages.TextualEvent',
+};
```

Comment:
As per https://matrix.org/docs/spec/client_server/r0.6.1#rich-replies they are only valid to `m.room.message` events

Reply:
They are still used and supported in other clients, I think maybe showing "`<click to jump to event>`" would at least help, even if that event is "unsupported".

Reply:
Yeah, even if the spec is bad, it doesn't hurt to try to render replies properly, since such replies can appear and do exist in rooms.

Reply:
In other spec-questionable places, Element happily renders nothing, it'd be nice to strive for consistency